pkg/node/node.go

```diff
@@ -955,9 +956,15 @@ func NewBee(interrupt chan struct{}, sysInterrupt chan os.Signal, addr string, p
 		return nil, fmt.Errorf("pullsync protocol: %w", err)
 	}
 
+	staking := staking.New(overlayEthAddress, chainCfg.StakingContract, erc20Address, transactionService, common.BytesToHash(nonce))
```

Comment:
I would prefer to use a different variable name here. This clashes with the package name.